File: java/arrays/second_largest.java
import java.util.Scanner;

public class SecondLargest {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter size of array (>=2): ");
        int n = sc.nextInt();

        if (n < 2) {
            System.out.println("Array must have at least 2 elements!");
            return;
        }

        int[] arr = new int[n];

        System.out.println("Enter " + n + " elements:");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        int first = Integer.MIN_VALUE;
        int second = Integer.MIN_VALUE;

        for (int i = 0; i < n; i++) {
            if (arr[i] > first) {
                second = first;
                first = arr[i];
            } else if (arr[i] > second && arr[i] != first) {
                second = arr[i];
            }
        }

        if (second == Integer.MIN_VALUE) {
            System.out.println("No second largest element (all elements equal).");
        } else {
            System.out.println("Second largest element: " + second);
        }
    }
}

File: python/arrays/second_largest.py
n = int(input("Enter size of array (>=2): "))
if n < 2:
    print("Array must have at least 2 elements!")
    exit()

arr = list(map(int, input(f"Enter {n} elements: ").split()))

first = second = float('-inf')
for num in arr:
    if num > first:
        second = first
        first = num
    elif num > second and num != first:
        second = num

if second == float('-inf'):
    print("No second largest element (all elements equal).")
else:
    print("Second largest element:", second)
